syntax = "proto3";

import "common.proto";

package erebus;

option optimize_for = LITE_RUNTIME;


service SystemInfo {
    rpc GetSystemInfo(SystemInfoRequest) returns(stream Property) {}
    rpc Ping(PingMessage) returns(PingMessage) {}
}

message SystemInfoRequest {
    string propertyNamePattern = 1;
}

message PingMessage {
    uint64 timestamp = 1;
    uint64 sequence = 2;
    bytes payload = 3;
}